globus_cli/parsing/shell_completion.py

```diff
@@ -0,0 +1,195 @@
+import os
+import shlex
+import click
+
+from globus_cli.safeio import safeprint
+from globus_cli.parsing.hidden_option import HiddenOption
+from globus_cli.parsing.case_insensitive_choice import CaseInsensitiveChoice
+
+SUPPORTED_SHELLS = ['BASH', 'ZSH']
+
+
+def safe_split_line(inputline):
+    parser = shlex.shlex(inputline, posix=True)
```

Comment:
Out of curiosity, is there anything we can do with this to make subshells get parsed correctly? Right now, if I use a subshell in the middle of a line (e.g. for `globus bookmark locate`) it breaks completion for the rest of the line after I finish my subshell.